### DISCUSSION TITLE
Add a comparison between LMQL and Guidance

### DISCUSSION THREAD

**Discussion by jvhoffbauer** [guidance-ai/guidance#199] [Q&A]
I would like to understand how Guidance compares to [LMQL](https://lmql.ai/) when reading the docs. Can we gather a list of additional features?

**Comment by slundberg** [comment]
People should feel free to leave notes here. The core of Guidance was written before LMQL (but not widely released), so the design choices were made independently and I expect there is stuff to learn beyond just the fundamental difference of a template language style vs. a SQL database style.

**Comment by jvhoffbauer** [comment]
Makes sense! Happy to start creating a list. 

Do I understand correctly that LMQL especially does not allow pattern enforcing in inference?

**Reply by marcotcr** [reply]
We are not keeping track of LMQL, so we don't know what they do or do not support :)

**Reply by charles-dyfis-net** [reply]
@jvhoffbauer, my understanding is that LMQL can enforce any constraint that can be represented as a context-free grammar.

**Reply by ogencoglu** [reply]
> We are not keeping track of LMQL, so we don't know what they do or do not support :)

Of course you do! Let's don't kid ourselves. Anyway, here is a comparison by LMQL: https://docs.lmql.ai/en/latest/python/comparison.html

**Comment by ianarawjo** [comment]
I was also curious about this. Given that guidance came later, it appeared to me, and other people as well, as a kind of knock-off of LMQL, except w a big corporation behind it. Maybe this perception is wrong —but, still, it would be nice to have a comparison somewhere to LMQL and pay respect where respect is due to the initial originators of the idea.

**Comment by ogencoglu** [comment]
LMQL is more open and less dismissive about such comparison and they provided a documentation related to that: https://docs.lmql.ai/en/latest/python/comparison.html
